### ISSUE TITLE
Reading scalar var fails with async

### ISSUE THREAD

**Issue by edhartnett** [NCAR/ParallelIO#966]
Reading a scalar var fails with async.

This is related to issue #204.

**Comment by jayeshkrishna** [comment]
Does reading scalar values work for non-async cases?

**Comment by edhartnett** [comment]
Sorry I took so long to answer your question.

Scalar vars worked with non-async cases, but only because gcc allows arrays to be declared with a size of 0. The reason for this is given here: https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html. It is a legacy way to support the "flexible array member" feature, which is new to me. This feature is supported differently (i.e. without a 0 length) in C99, but the gcc extension was there first.

For the curious, a good explanation of the flexible array member (which we don't use anywhere in our code) can be found here:
http://stackoverflow.com/questions/11733981/what-is-the-purpose-of-a-zero-length-array-in-a-struct

With the PR I merged this morning, scalar vars are now read without arrays of 0 length, as a side effect of other changes I made. So that point is moot.

I have some test code I just put up for PR that demonstrates that scalar vars are working for non-async, for all types (including netCDF-4 extended types).

I am working on pnetcdf async scalar vars on another issue (#968).
